M2/Macaulay2/packages/Permutations/docs.m2

```diff
@@ -0,0 +1,1296 @@
+    Example
+      p = permutation {3,1,2}
+      A = matrix {{1,2},{3,4},{5,6}}
+      p * A
+///
+
+-- (symbol *, Permutation, Permutation)
+doc ///
+  Key
+    (symbol *, Permutation, Permutation)
+  Headline
+    computes the product of two permutations
+  Usage
+    w * v
+  Inputs
+    w:Permutation
+    v:Permutation
+  Outputs
+    :Permutation
+  Description
+    Text
+      The product of two permutations $p$ and $q$ is given by their composition,
+      i.e., $ (p \circ q)(i) = p(q(i))$.
+    Example
+      p = permutation {3,1,2,5,4}
+      q = permutation {2,1,3,4,5}
+      p * q
+    Text
+      The two permutations do not need to be the same length.
+    Example
+      p = permutation {3,1,2,5,4}
+      q = permutation {2,1,3}
+      p * q
+///
+
+-- (symbol ^, Permutation, ZZ)
+doc ///
+  Key
+    (symbol ^, Permutation, ZZ)
+  Headline
+    computes the power of a permutation
+  Usage
+    w^n
+  Inputs
+    w:Permutation
+    n:ZZ
+  Outputs
+    :Permutation
+  Description
+    Text
+      Computes the power of a permutation. The power can be any integer.
+    Example
+      p = permutation {3,1,2,5,4}
+      p^2
+      p^0
+      p^(-3)
+///
+
+-- ascendingRuns
+doc ///
+  Key
+    ascendingRuns
+    (ascendingRuns, Permutation)
+  Headline
+    computes the ascending runs of a permutation
+  Usage
+    ascendingRuns w
+  Inputs
+    w:Permutation
+  Outputs
+    allRuns:List
+  Description
+    Text
+      An \emph{ascending run} is a maximal subsequence of successive ascents.
+    Example
+      p = permutation {3,1,2,5,4}
+      ascendingRuns p
+  SeeAlso
+    ascents
+    descendingRuns
+    descents
+    exceedances
+    saliances
+    records
+///
+
+-- ascents
+doc ///
+  Key
+    ascents
+    (ascents, Permutation)
+  Headline
+    computes the ascents of a permutation
+  Usage
+    ascents w
+  Inputs
+    w:Permutation
+  Outputs
+    :List
+  Description
+    Text
+      A permutation $p=(p_1 \, \hdots \, p_n)$ has an \emph{ascent} at $i$ (for $i < n$)
+      if $p(i) < p(i+1)$.
+    Example
+      p = permutation {3,1,2,5,4}
+      ascents p
+  SeeAlso
+    ascendingRuns
+    descendingRuns
+    descents
+    exceedances
+    saliances
+    records
+///
+
+-- avoidsPattern
+doc ///
+  Key
+    avoidsPattern
+    (avoidsPattern, Permutation, List)
+  Headline
+    whether a permutation avoids a pattern
+  Usage
+    avoidsPattern(w, pattern)
+  Inputs
+    w:Permutation
+    pattern:List
+  Outputs
+    isAvoiding:Boolean
+  Description
+    Text
+      Pattern avoidance is more easily understood through an example.
+      A permutation $p$ is $2143$-avoiding if there do not exist indices 
+      $i < j < k < l$ such that $w_j < w_i < w_l < w_k$.
+    Example
+      p = permutation {3,1,2,5,4}
+      avoidsPattern(p, {2,1,4,3})
+  SeeAlso
+    avoidsPatterns
+    isCartwrightSturmfels
+    isCDG
+    isVexillary
+///
+
+-- avoidsPatterns
+doc ///
+  Key
+    avoidsPatterns
```

Comment:
Under `SeeAlso`, you can link to `"MatrixSchubert::avoidsAllPatterns"` and `"MatrixSchubert::isPatternAvoiding"`.